hash_practice/exercises.py

```diff
@@ -2,28 +2,98 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+
+    anagram_list = []
+
+    word_dict = {}
+    for word in strings:
+        sorted_word = "".join(sorted(word))
+        if sorted_word in word_dict:
+            word_dict[sorted_word].append(word)
+        else:
+            word_dict[sorted_word] = [word]
+
+    for value in word_dict.values():
+        anagram_list.append(value)
+
+    return anagram_list
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n) (This may actually be O(n^2) because of the max function on line 46, but I'm not sure)
+        Space Complexity: O(n)
     """

```

Comment:
👍 I would say that the time complexity is O(nk) since the loop on line 43 goes k times.